### PR TITLE
[feat] Add Keycloak Auth Resource (Async/Sync) Clients & Tests  

### PR DESCRIPTION
**Pull request by leebongseung**
## 📝 Work Summary
- [x] Added `KeycloakAuthResourceAsyncClient` and `KeycloakAuthResourceClient` interfaces for CRUD operations.  
- [x] Implemented `KeycloakAuthResourceAsyncClientImpl` and `KeycloakAuthResourceClientImpl`.  
- [x] Extended `ClientConfiguration` with `getAuthResourceUrl` and `getAuthResourceSearchUrl`.  
- [x] Added `ResourceQueryParams` DTO with builder, typed fields, and URL encoding support.  
- [x] Integrated new clients into `KeycloakClient` factory.  
- [x] Implemented comprehensive async CRUD test suite (`KeycloakAuthResourceAsyncClientImplTest`).  
- [x] Added dedicated unit tests for `ResourceQueryParams`.  
- [x] Updated `build.gradle` to enforce UTF-8 encoding for tests.  

---

## 💡 Review Points
1. Verify that async and sync client implementations (`block()` usage in sync) follow best practices.  
2. Confirm that `ResourceQueryParams` handles edge cases (null, empty, special characters, UUID).  
3. Ensure URL encoding logic is consistent with Keycloak API expectations.  
4. Validate test coverage: success, failure, and boundary cases are sufficiently covered.  
5. Review potential performance implications of deep + exactName queries with multiple filters.  

---

## 📂 Changed Files & Description
1. `build.gradle`: Enforce UTF-8 for test JVM.  
2. `KeycloakAuthResourceAsyncClient.java`: New async client interface for auth resource CRUD.  
3. `KeycloakAuthResourceAsyncClientImpl.java`: Implementation of async client.  
4. `KeycloakAuthResourceClient.java`: New sync client interface.  
5. `KeycloakAuthResourceClientImpl.java`: Sync implementation delegating to async client.  
6. `ClientConfiguration.java`: Added auth resource URL builders.  
7. `ResourceQueryParams.java`: DTO for query parameters with encoding logic.  
8. `KeycloakClient.java`: Integrated new async/sync auth resource clients.  
9. `KeycloakAuthResourceAsyncClientImplTest.java`: Comprehensive CRUD + query tests.  
10. `ResourceQueryParamsTest.java`: Unit tests for query param encoding and composition.  

---

## 🚨 Other Notes
- The async tests rely on a live Keycloak container (`KeycloakShareTestContainer`). Ensure client UUIDs and tokens are configured properly.  
- Boundary cases (negative pagination, empty name, special characters) are tested but may behave differently depending on Keycloak server implementation.  
- Consider adding integration tests with real Keycloak to validate end-to-end behavior.

### CODE REVIEW

File: src/main/java/com/sd/KeycloakClient/client/admin/auth/async/impl/KeycloakAuthResourceAsyncClientImpl.java [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+   }
+
+   @Override
+   public Mono<KeycloakResponse<ResourceRepresentation>> getResource(
+       String accessToken,
+       UUID clientUuid,
+       UUID resourceId
+   ) {
+      return http.<ResourceRepresentation>get(
+              configuration.getAuthResourceUrl(Objects.toString(clientUuid, ""), Objects.toString(resourceId, "")))
```
Comment (sangwonyu) [inline src/main/java/com/sd/KeycloakClient/client/admin/auth/async/impl/KeycloakAuthResourceAsyncClientImpl.java:30]: clientUuid, resourceId이 필수 값일까요?
url 생성되는거 보면, //clientId
이런식으로 만들어질거같은데, 이렇게 되면 어차피 키클락쪽에서 에러날거같은데 

빈값을 넣어주기보단 키클락까지 까지 않고 바로 필수값 메세지를 리턴해주는게 어떨까요

File: src/test/java/com/sd/KeycloakClient/client/admin/auth/async/impl/KeycloakAuthResourceAsyncClientImplTest.java [hunk 2]
```diff
@@ -0,0 +1,787 @@
...
+                   assertThat(body.getName()).isEqualTo(name);
+                })
+                .verifyComplete();
+         }
+      }
+
+      @Nested
+      @TestInstance(TestInstance.Lifecycle.PER_CLASS)
+      @DisplayName("Failure Cases")
+      class FailureCases {
```
Comment (sangwonyu) [inline src/test/java/com/sd/KeycloakClient/client/admin/auth/async/impl/KeycloakAuthResourceAsyncClientImplTest.java:106]: clientId, resourceId이 정상적이지 않은 값을 넣었을 때도 테스트 케이스에 넣으면 좋을거같아요

File: src/test/java/com/sd/KeycloakClient/client/admin/auth/async/impl/KeycloakAuthResourceAsyncClientImplTest.java [hunk 3]
```diff
@@ -0,0 +1,787 @@
...
+         static Stream<String> nameProvider() {
+            return Stream.of("res-한글", "res with space", "res/with/slash", "a+b=c&d");
+         }
+
+         @ParameterizedTest(name = "case1.{index}: name=\"{0}\" & exactName=true -> 200 & all match")
+         @MethodSource("nameProvider")
+         @DisplayName("case1: name + exactName=true")
+         void byName_exact_true(String name) {
+            //given
+            createResourceAndGetId(name, null, null);
```
Comment (sangwonyu) [inline src/test/java/com/sd/KeycloakClient/client/admin/auth/async/impl/KeycloakAuthResourceAsyncClientImplTest.java:186]: Resource를 몇개 더 넣고 테스트 하면 좋을거같아요.
name이 비슷한 값이 여러개 있어도 extractName에 의해 정확한 값만 매칭돼서 출력되는지를  파악해주는 식의 테스트 코드를 짜야 명확해서 좋을거같아요.
( 다른 부분 테스트 부분도 동일. 하나의 값만 가지고 테스트하니 정확한 테스트 증명이 안될 수 있음.) 

계속 생성 시키는게 부담되면, 키클락 json 파일에 resource넣을 수 있으면 넣고 테스트해도 괜찮을거같네요.

File: src/test/java/com/sd/KeycloakClient/client/admin/auth/async/impl/KeycloakAuthResourceAsyncClientImplTest.java (hunk 2 above)
Comment (leebongseung) [reply src/test/java/com/sd/KeycloakClient/client/admin/auth/async/impl/KeycloakAuthResourceAsyncClientImplTest.java:106]: 반영하도록 하겠습니다!

File: src/test/java/com/sd/KeycloakClient/client/admin/auth/async/impl/KeycloakAuthResourceAsyncClientImplTest.java (hunk 3 above)
Comment (leebongseung) [reply src/test/java/com/sd/KeycloakClient/client/admin/auth/async/impl/KeycloakAuthResourceAsyncClientImplTest.java:186]: 접두 확장: a, ab, abc, abcd
접미 변형: report, report1, report2
공백/특수문자 변형: report, report v1, report-v1, report_v1
대소문자 변형: Report, REPORT, report (Keycloak 검색이 case-sensitive 한지 확인 여부 포함)

→ 위 케이스들을 포함해서 테스트 보강하겠습니다.

File: src/main/java/com/sd/KeycloakClient/client/admin/auth/async/impl/KeycloakAuthResourceAsyncClientImpl.java (hunk 1 above)
Comment (sangwonyu) [reply src/main/java/com/sd/KeycloakClient/client/admin/auth/async/impl/KeycloakAuthResourceAsyncClientImpl.java:30]: 아주 좋아유~

**Review by sangwonyu — APPROVED** [review]
